inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/sidebar/RecommendationSidebar.java

```diff
@@ -67,5 +91,98 @@ public RecommendationSidebar(String aId, IModel<AnnotatorState> aModel,
                 aAnnotationPage.actionRefreshDocument(_target)));
         
         add(form);
-    }
+
+		chartContainer = new Label("chart-container" );
+		add(chartContainer);
+
+		add(new AbstractAjaxTimerBehavior(Duration.milliseconds(5000)) {
+			private static final long serialVersionUID = -3782208159226605584L;
+
+			@Override
+			protected void onTimer(AjaxRequestTarget target) {
+
+				List<LoggedEvent> loggedEvents = recommendationService.listLoggedEvents(aModel.getObject().getProject(),
+						aModel.getObject().getUser());
+
+				// iterate over the logged events to extract the scores only.
+				List<Double> listDetailJson = new ArrayList<Double>();
+
+				for (LoggedEvent loggedEvent : loggedEvents) {
+					String detailJson = loggedEvent.getDetails();
+
+					ObjectMapper mapper = new ObjectMapper();
+					Double scoreDouble;
+					try {
+						LoggedEventDetail detail = mapper.readValue(detailJson, LoggedEventDetail.class);
+
+						try {
+							scoreDouble = Double.valueOf(detail.getScore());
+							listDetailJson.add(scoreDouble);
+						} catch (NumberFormatException e) {
+							log.debug("Skipping logged Event due to invalid score, Detail:{}", detail);
+							continue;
+						}
+
+					} catch (IOException e) {
+						log.error(e.toString(), e);
+					}
+				}
+
+				String data = "";
+				for (Double integer : listDetailJson) {
+					data+= ",";
+					data += integer;
+				}
+
+				String javascript = "var chart=c3.generate({bindto:\"#chart-container-js\",data:{columns:[[\"data1\""+data +"]]}});;";
+				target.prependJavaScript(javascript);
+			}
+		});
+	}
+
+	private String getJson() {
+		// TODO Auto-generated method stub
+		return "";
+	}
+
+	@Override
+	protected void onAfterRender() {
+		super.onAfterRender();
+	}
+
+	@Override
+	public void renderHead(IHeaderResponse aResponse) {
+		super.renderHead(aResponse);
+
+		//import Css
+		aResponse.render(JavaScriptHeaderItem.forReference(C3JsReference.get()));
+		aResponse.render(JavaScriptHeaderItem.forReference(D3JsReference.get()));
+
+		//import Css
+		aResponse.render(CssHeaderItem.forReference(C3CssReference.get()));
+	}
+}
+
+//TODO: Move it to a separate file?
+@JsonIgnoreProperties(ignoreUnknown = true)
```

Comment:
Let's re-use the class `de.tudarmstadt.ukp.inception.recommendation.log.RecommenderEvaluationResultEventAdapter.Details` instead of defining a new one.